src/components/form/AddPlant.tsx

```diff
@@ -0,0 +1,128 @@
+import { ChangeEvent, FormEvent, useContext, useState } from 'react';
+import { plantBotanicNames } from '../../data/plantBotanicNames';
+import { rooms } from '../../data/rooms';
+import sprite from '../../icons/symbol-defs.svg';
+import { PlantsList } from '../App';
+
+type AddPlantProps = {
+  onClose: () => void;
+};
+
+const AddPlant: React.FC<AddPlantProps> = ({ onClose }) => {
+  const [selectedCommonName, setSelectedCommonName] = useState<string>('');
+  const [selectedBotanicName, setSelectedBotanicName] = useState<string>(
+    plantBotanicNames[0]
+  );
+  const [selectedRoom, setSelectedRoom] = useState<string>(rooms[0]);
+
+  const plantsListData = useContext(PlantsList);
+
+  const handleSelect = (
+    e: ChangeEvent<HTMLSelectElement | HTMLInputElement>
+  ): void => {
+    const { name } = e.target;
+    switch (name) {
```

Comment:
bad practice to leave the default case empty, you can use it for one of the cases you have or at least throw an error or make console.warn so if something go wrong you gonna know what happened
IMHO :)